module/zfs/vdev_label.c

```diff
@@ -500,6 +500,11 @@ vdev_config_generate(spa_t *spa, vdev_t *vd, boolean_t getstats,
 			fnvlist_add_uint64(nv, ZPOOL_CONFIG_VDEV_TOP_ZAP,
 			    vd->vdev_top_zap);
 		}
+
+		if (vd->vdev_resilver_defered) {
+			ASSERT(vd->vdev_ops->vdev_op_leaf);
```

Comment:
could we also assert that `spa_resilver_deferred` is set?

Reply:
Sure. I will add it.